### ISSUE TITLE
problems of selectedPics()

### ISSUE THREAD

**Issue by taehee28** [ShahabGT/InstagramPicker#7]
hi, i'm trying to using your good Image Picker

I found two problem 

one, 
when single or multiple image is finally picked and get back to original activity, selectPics() is called
but in situation that call ImagePicker Activity not first(like, use ImagePicker again),  addresses parameter of selectPics() is not cleared
addresses has data of images already picked before...
it have to clear list to write code??

two,
selectPics() called multiple times when get back to original activity
'how many called' depends on addresses's size
 why it happens??

waiting for answer

**Comment by ShahabGT** [comment]
hi there,
Thanks for using my library,
would you please send your code here to check what might be the problem?
thanks.

**Comment by taehee28** [comment]
Thanks for answer!

Here's my code


```
// Click the Add Button, addPhoto() is called

private void addPhoto() {

    InstagramPicker in = new InstagramPicker(this);
    in.show(1, 1, 10, new MultiListener() {

        @Override
        public void selectedPics(List<String> addresses) {

                Log.d("testlog", addresses.size()+"");

                // set imageview with picked image and attach to layout...

        }

    });

}
```

I really nothing to do..
Deleted every code in `selectedPics()` without `Log.d()`, it happens again

```
2020-09-23 09:11:44.737 25282-25282/com.xxx D/testlog: 1
2020-09-23 09:12:04.974 25282-25282/com.xxx D/testlog: 2
2020-09-23 09:12:05.023 25282-25282/com.xxx D/testlog: 2
```
First line of logs printed when got back from ImagePicker Activity very first time (picked single image)
Second and third line of logs printed  when got back from ImagePicker Activity second time. (also picked single image, I used ImagePicker only once at that time) 
In second time, `adresses` list had image data of first picked and second picked, and `selectedPics()` called twice

Now I tested the code several times, it just see the printed logs though...
I said 'how many called `selectedPics()`' depends on size of `adresses` before. But I found it isn't not really depends on size
```
2020-09-23 09:29:39.739 26843-26843/com.xxx D/testlog: 3
2020-09-23 09:29:39.739 26843-26843/com.xxx D/testlog: 3
2020-09-23 09:29:39.739 26843-26843/com.xxx D/testlog: 3
2020-09-23 09:29:39.739 26843-26843/com.xxx D/testlog: 3
```
4 lines of logs are printed same time... it means `selectedPics()` called four times

Is there any faults in my code? or am I just misunderstand usage?

**Comment by ShahabGT** [comment]
fixed in version 2.1.1
